### ISSUE TITLE
Typing '1' or '2' into an input inside a web component toggles the content/info content apps instead of writing '1' or '2'

### ISSUE THREAD

**Issue by JasonElkin** [umbraco/Umbraco-CMS#13838]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.4.0

### Bug summary

I wrapped a library up a web component in the v10 backoffice but can't type 1 or 2 into it. This issue is the same as #6508 but #7186 doesn't fix it.

### Steps to reproduce
The issue is _technically_ the same as this:
Create a simple property editor, or modify an existing one, and add the following example UUI component:
```
<uui-input
  label="Username"
  type="text"
  placeholder="Enter your username..."></uui-input>
```

Type 2 into the input, and you'll be taken to the "Info" content app.
Type 1 and you'll be taken back to "Content"

Neither 1 or 2 will appear in the input.

I _could_ add disable-hotkeys which gets around the issue when the use case is simple like the above, but I actually have a much more complex structure that looks like this:

```
<jasons-magical-web-component>
    <third-party-web-component>
       <!-- shadow DOM, here be dragons... maybe... I don't know 🤷-->
       <input-web-component> <!-- This is the actual event target for the keydown, I can't add the "disable-hotkeys" attribute here --> 
       </input-web-component>
    </third-party-web-component>
</jasons-magical-web-component>
```

### Expected result / actual result

Typing 1 or 2 with the input selected, adds a 1 or 2 character into the input.

---
_This item has been added to our backlog [AB#26831](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/26831)_

**Comment by elit0451** [comment]
Hi @JasonElkin 👋 

Thanks for reporting this. You are right, I was able to reproduce the issue. However, I managed to make it work by putting `disable-hotkeys` in the `<uui-input>`, like:

```html
<uui-input disable-hotkeys
	label="Username"
	type="text"
	placeholder="Enter your username...">
</uui-input>
```

I need some additional input from our front-end team about that though, so I will get back to you soon 🙂

**Comment by JasonElkin** [comment]
Thanks @elit0451, yes I just realised that works - my actual use case was more complex, which this doesn't help with. I'll edit the issue with more details.

**Comment by JasonElkin** [comment]
As a workaround, I've updated my outer component with something a bit like this:
```
const digits = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];

@customElement('jasons-magical-web-component')
export class JasonsMagicalWebComponent extends LitElement {
	constructor() {
		super();
		this.addEventListener('keydown', (e) => {
			if (e.ctrlKey === e.metaKey === e.altKey === false && e.key in digits) {
				e.stopPropagation();
			}
		})
	}
	// ... component stuff
}
```

WRT a more helpful fix, and to avoid duplicating this kind of code, this kind of thing _could_ be encapsulated in its own web component...
```
<disable-hotkeys/>
  <jasons-magical-web-component></jasons-magical-web-component>
</disable-hotkeys>
```
Or even, with some smarter logic:
```
<disable-hotkeys keys="[1, 2, 3, { 'ctrl', 'alt', 's' }]"/>
  <jasons-magical-web-component></jasons-magical-web-component>
</disable-hotkeys>
```

**Comment by elit0451** [comment]
@JasonElkin could you try and see if adding this would fix your problem `ng-on-input="$event.stopPropagation()"`?

**Comment by elit0451** [comment]
Never mind, you've tried something similar as a workaround 😅 I will then mark this as up for grabs where the fix that we would be aiming for is to turn the logic you've put as your workaround into an AngularJS directive that we can use across 🙂

**Comment by markadrake** [comment]
I noticed the code already checks from some common-sense sources in which the hotkey should not trigger.

The only drawback I could find is that most web components will take advantage of the shadow-dom. I've added a check to the code to look at the event's `composedPath`. If the shadow-dom is not closed, this can provide us an opportunity to catch more edge cases. 

I tested this change in Chrome and Firefox. I added a new property editor, a `uui-input` _without_ the `disable-hotkeys` attribute. Worked like a charm.

This will have no meaningful effect if your shadow-dom is closed.
